addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/discovery/AVMFritzUpnpDiscoveryParticipant.java

```diff
@@ -33,27 +40,43 @@
  *
  * @author Robert Bausdorf - Initial contribution
  * @author Christoph Weitkamp - Added support for groups
+ * @author Christoph Weitkamp - Use "discovery.avmfritz:background=false" to disable discovery service
  */
-@Component(service = UpnpDiscoveryParticipant.class, immediate = true)
+@Component(service = UpnpDiscoveryParticipant.class, immediate = true, configurationPid = "discovery.avmfritz")
+@NonNullByDefault
 public class AVMFritzUpnpDiscoveryParticipant implements UpnpDiscoveryParticipant {
-    /**
-     * Logger
-     */
+
     private final Logger logger = LoggerFactory.getLogger(AVMFritzUpnpDiscoveryParticipant.class);
 
-    /**
-     * Provide supported ThingTypeUIDs
-     */
+    private boolean isAutoDiscoveryEnabled = true;
+    private Set<ThingTypeUID> supportedThingTypes = SUPPORTED_BRIDGE_THING_TYPES_UIDS;
+
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    @Modified
+    protected void modified(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    private void activateOrModifyService(ComponentContext componentContext) {
+        Dictionary<String, @Nullable Object> properties = componentContext.getProperties();
+        String autoDiscoveryPropertyValue = (String) properties.get("background");
+        if (autoDiscoveryPropertyValue != null && autoDiscoveryPropertyValue.length() != 0) {
+            isAutoDiscoveryEnabled = Boolean.valueOf(autoDiscoveryPropertyValue);
+        }
+        supportedThingTypes = isAutoDiscoveryEnabled ? SUPPORTED_BRIDGE_THING_TYPES_UIDS : Collections.emptySet();
+    }
+
     @Override
     public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
-        return SUPPORTED_BRIDGE_THING_TYPES_UIDS;
+        return supportedThingTypes;
     }
 
-    /**
-     * Create a discovery result from UPNP discovery
-     */
     @Override
-    public DiscoveryResult createResult(RemoteDevice device) {
+    public @Nullable DiscoveryResult createResult(RemoteDevice device) {
         ThingUID uid = getThingUID(device);
```

Comment:
Would applying the `isAutoDiscoveryEnabled` here make more sense?

Reply:
@martinvw See #3785